instrumentation/net/http/otelhttp/transport.go

```diff
@@ -32,12 +35,18 @@ import (
 type Transport struct {
 	rt http.RoundTripper
 
-	tracer            trace.Tracer
-	propagators       propagation.TextMapPropagator
-	spanStartOptions  []trace.SpanStartOption
-	filters           []Filter
-	spanNameFormatter func(string, *http.Request) string
-	clientTrace       func(context.Context) *httptrace.ClientTrace
+	tracer                trace.Tracer
+	meter                 metric.Meter
+	propagators           propagation.TextMapPropagator
+	spanStartOptions      []trace.SpanStartOption
+	readEvent             bool
+	filters               []Filter
+	spanNameFormatter     func(string, *http.Request) string
+	clientTrace           func(context.Context) *httptrace.ClientTrace
+	getRequestAttributes  func(*http.Request) []attribute.KeyValue
+	getResponseAttributes func(response *http.Response) []attribute.KeyValue
+	counters              map[string]metric.Int64Counter
+	valueRecorders        map[string]metric.Float64Histogram
```

Comment:
Why are we using maps here? How about simply having the mutiple instrument fields (many `metric.Int64Counter`  and `metric.Float64Histogram` fields instead of these two maps).

Reply:
Also the same thing done by [the server handler](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/handler.go#L46), this pattern seems to be used in a lot of instrumentations.